include/sparrow/array/array_data.hpp

```diff
@@ -99,6 +135,75 @@ namespace sparrow
         friend class iterator_access;
     };
 
+    /***********************************
+     * getter functions for array_data *
+     ***********************************/
+
+    inline data_descriptor type_descriptor(const array_data& data)
+    {
+        return data.type;
+    }
+
+    inline array_data::length_type length(const array_data& data)
+    {
+        return data.length;
+    }
+
+    inline std::int64_t offset(const array_data& data)
+    {
+        return data.offset;
+    }
+
+    inline array_data::bitmap_type& bitmap(array_data& data)
+    {
+        return data.bitmap;
+    }
+
+    inline const array_data::bitmap_type& bitmap(const array_data& data)
+    {
+        return data.bitmap;
+    }
+
+    inline std::size_t buffers_size(const array_data& data)
+    {
+        return data.buffers.size();
+    }
+
+    inline array_data::buffer_type& buffer_at(array_data& data, std::size_t i)
+    {
+        return data.buffers[i];
```

Comment:
We might want to add assertions in accessors which take a size, that will be helpful.